### PR TITLE
Elisha - Migrate transfers

### PR DESCRIPTION
**Pull request by SweetmanTech**

### CODE REVIEW

File: src/Transfers.sol [hunk 1]
```diff
@@ -2,15 +2,22 @@
 pragma solidity ^0.8.15;
 
 import {Cre8orsERC6551} from "./utils/Cre8orsERC6551.sol";
+
+import {Cre8ing} from "./Cre8ing.sol";
 import {ICre8ors} from "./interfaces/ICre8ors.sol";
+import {ICre8ing} from "./interfaces/ICre8ing.sol";
 import {IERC721Drop} from "./interfaces/IERC721Drop.sol";
 
+import "forge-std/console.sol";
```
Comment (SweetmanTech) [inline src/Transfers.sol:11]: remove consoles before merging
```suggestion
```

File: src/Transfers.sol [hunk 2]
```diff
@@ -2,15 +2,22 @@
 pragma solidity ^0.8.15;
 
 import {Cre8orsERC6551} from "./utils/Cre8orsERC6551.sol";
+
```
Comment (SweetmanTech) [inline src/Transfers.sol:5]:
```suggestion
```

File: test/Transfers.t.sol [hunk 3]
```diff
@@ -0,0 +1,148 @@
+import {TransferHook} from "../src/Transfers.sol";
+
```
Comment (SweetmanTech) [inline test/Transfers.t.sol:2]: any reason you've got empty lines between your imports?
```suggestion
```

**Review by SweetmanTech — COMMENTED** [review]
getting really close

File: test/Cre8ing.t.sol [hunk 4]
```diff
@@ -8,17 +8,37 @@ import {DummyMetadataRenderer} from "./utils/DummyMetadataRenderer.sol";
...
     address public constant DEFAULT_CRE8OR_ADDRESS = address(456);
     address public constant DEFAULT_TRANSFER_ADDRESS = address(0x2);
 
     function setUp() public {
         Cre8orTestBase.cre8orSetup();
         cre8ingBase = new Cre8ing();
-        vm.prank(DEFAULT_OWNER_ADDRESS);
+        transferHook = new TransferHook();
+        vm.startPrank(DEFAULT_OWNER_ADDRESS);
         cre8orsNFTBase.setCre8ing(cre8ingBase);
```
[the remark this answers is not in the document]
Comment (Bulalu) [reply test/Cre8ing.t.sol]: removed

File: src/minter/FriendsAndFamilyMinter.sol [hunk 5]
```diff
@@ -39,7 +41,11 @@ contract FriendsAndFamilyMinter is IFriendsAndFamilyMinter {
         hasDiscount[recipient] = false;
 
         // Set lockup information (optional)
-        ILockup lockup = ICre8ors(cre8orsNFT).cre8ing().lockUp(cre8orsNFT);
+        ILockup lockup = ITransfers(
+            ICre8ors(cre8orsNFT).getHook(
+                IERC721ACH.HookType.BeforeTokenTransfers
+            )
+        ).cre8ing().lockUp(cre8orsNFT);
```
Comment (crazzywizard) [inline src/minter/FriendsAndFamilyMinter.sol:48]: this needs to be changed. This is an error in the contract that is fixed in an open pr

**Review by crazzywizard — APPROVED** [review]
One change needed

File: src/minter/FriendsAndFamilyMinter.sol (hunk 5 above)
Comment (SweetmanTech) [reply src/minter/FriendsAndFamilyMinter.sol:48]: better now?

**Review by crazzywizard — APPROVED** [review]

**Pull request closed without merging**